TS05-Plotting-Time-Series-ggplot-In-R.Rmd

```diff
@@ -141,108 +133,258 @@ harTemp.monthly.09.11$date <- as.POSIXct(harTemp.monthly.09.11$datetime)
+to the plot object `AirTempDaily` that we just created. 
+
+```{r format-x-axis-labels }
+#format x-axis: dates
+AirTempDailyb <- AirTempDaily + 
+  (scale_x_datetime(labels=date_format("%b %y")))
+
+AirTempDailyb
+```
+
+###Adjust Date Ticks
+
+We can adjust the date ticks too. In this instance, having 1 tick per year may be
+enough. If we have the `scales` package loaded, we can use 
+`breaks=date_breaks("1 year")` to create a tick for every year.
+We can adjust this as needed (e.g. 10 days, 30 days, 1 month, etc).
+
+```{r format-x-axis-label-ticks }
+#format x-axis: dates
+AirTempDailyb <- AirTempDaily + 
+    (scale_x_datetime(breaks=date_breaks("6 months"),
+      labels=date_format("%b %y")))
+
+AirTempDailyb
+
 #format x-axis: dates
-AirTempDailyb <- AirTempDailya + (scale_x_datetime(labels=date_format("%b %y")))
+AirTempDailyb <- AirTempDaily + 
+    (scale_x_datetime(breaks=date_breaks("1 year"),
+      labels=date_format("%b %y")))
+
 AirTempDailyb
+
+
 ```
 
-The dates are now legible on the x-axis. Notice in the code for the x scale
-(`scale_x_datetime()`), we changed the date format from %m to %b which gives the
-abreviated month.
 
-Bonus: If an x variable is not a datetime class (e.g., not POSIX),
-`scale_x_...()` exist to help format x-axes. {.notice2}
+<i class="fa fa-star"></i> **Data Tip:**  We can adjust the tick spacing and format
+for x and y axes using `scale_x_continuous` or `scale_y_continuous` to format 
+a continue variable. Check out `??scale_x_` (tab complete to view the various x 
+and y scale options)
+{: .notice }
+
+##GGPLOT - Subset by Time
+
+Sometimes we want to zoom into a region of our plot without subsetting the 
+entire `data_frame`. To do this, we can define start and end times. We can then
+define the `limits` in the `scale_x_datetime` object as follows:
+
+`scale_x_datetime(limits=start.end) +`
 
-##Use Theme() to Customize ggplot Figures
-The ability to customize nearly every aspect of the plot is one of the reasons
-to use `ggplot2`. The `theme()` function is a very useful tool for changing the
-appearance of all of the non-data elements of a figure.  We are only going to 
-customize the text in this section.  You can find the other arguments for 
-`theme()` at Hadley Wickham's <a href="http://docs.ggplot2.org/0.9.2.1/theme.html" target="_blank"> 
-documentation</a> `theme()` documentation. 
+
+```{r subset-ggplot-time }
+
+#Define Start and end times for the subset as R objects that are the time class
+startTime <- as.POSIXct("2011-01-01 00:00:00")
+endTime <- as.POSIXct("2012-01-01 00:00:00")
+
+#create a start and end time R object
+start.end <- c(startTime,endTime)
+start.end
+
+#View data for 2011 only
+AirTempDailyb <- AirTempDaily + 
+    (scale_x_datetime(limits=start.end,
+                      breaks=date_breaks("1 year"),
+                      labels=date_format("%b %y")))
+
+AirTempDailyb
+
+```
+
+##GGPLOT Themes
```

Comment:
This is a title. I don't love lower case in titles.
Let's create an issue about libraries and cap rules. I get consistency but readability and typography consistency is important too.